src/bsoncxx/test/CMakeLists.txt

```diff
@@ -85,56 +85,58 @@ if(ENABLE_MACRO_GUARD_TESTS)
+            # bsoncxx/v1/config/config.hpp (generated by CMake)
             BSONCXX_POLY_USE_IMPLS
             BSONCXX_POLY_USE_STD
 
-            # version.hpp (generated by CMake)
+            # bsoncxx/v1/config/export.hpp (generated by CMake)
+            BSONCXX_ABI_CDECL
+            BSONCXX_ABI_EXPORT
+            BSONCXX_ABI_EXPORT_CDECL
+            BSONCXX_ABI_NO_EXPORT
+            BSONCXX_DEPRECATED
+
+            # bsoncxx/v1/config/version.hpp (generated by CMake)
+            BSONCXX_VERSION_STRING
             BSONCXX_VERSION_EXTRA
             BSONCXX_VERSION_MAJOR
             BSONCXX_VERSION_MINOR
             BSONCXX_VERSION_PATCH
 
-            # export.hpp (generated by CMake)
-            BSONCXX_ABI_CDECL
-            BSONCXX_ABI_EXPORT
-            BSONCXX_ABI_EXPORT_TESTING
-            BSONCXX_ABI_NO_EXPORT
-            BSONCXX_ABI_EXPORT_CDECL
-            BSONCXX_ABI_EXPORT_CDECL_TESTING
-            BSONCXX_DEPRECATED
+            # bsoncxx/v1/detail/macros.hpp
+            BSONCXX_PRIVATE_CONCAT
+            BSONCXX_PRIVATE_CONCAT_IMPL
+            BSONCXX_PRIVATE_CONSTEXPR_CXX14
+            BSONCXX_PRIVATE_FORCE_SEMICOLON
+            BSONCXX_PRIVATE_FWD
+            BSONCXX_PRIVATE_IF_CLANG
+            BSONCXX_PRIVATE_IF_GCC
+            BSONCXX_PRIVATE_IF_GNU_LIKE
+            BSONCXX_PRIVATE_IF_MSVC
+            BSONCXX_PRIVATE_PRAGMA
+            BSONCXX_PRIVATE_PRAGMA_IMPL
+            BSONCXX_PRIVATE_RETURNS
+            BSONCXX_PRIVATE_STRINGIFY
+            BSONCXX_PRIVATE_STRINGIFY_IMPL
+            BSONCXX_PRIVATE_UNREACHABLE
+            BSONCXX_PRIVATE_WARNINGS_DISABLE
+            BSONCXX_PRIVATE_WARNINGS_DISABLE_IMPL_FOR_Clang
+            BSONCXX_PRIVATE_WARNINGS_DISABLE_IMPL_FOR_GCC
+            BSONCXX_PRIVATE_WARNINGS_DISABLE_IMPL_FOR_GNU
+            BSONCXX_PRIVATE_WARNINGS_DISABLE_IMPL_FOR_MSVC
+            BSONCXX_PRIVATE_WARNINGS_POP
+            BSONCXX_PRIVATE_WARNINGS_PUSH
+
+            # bsoncxx/v1/detail/prelude.hpp
+            BSONCXX_PRIVATE_V1_INSIDE_MACRO_GUARD_SCOPE
 
-            # prelude.hpp
-            BSONCXX_UNREACHABLE
-
-            # util.hpp
-            _bsoncxxDisableWarningImpl_for_GCC
-            _bsoncxxDisableWarningImpl_for_GNU
-            _bsoncxxDisableWarningImpl_for_MSVC
-            _bsoncxxDisableWarningImpl_for_Clang
-            BSONCXX_PUSH_WARNINGS
-            BSONCXX_POP_WARNINGS
-            BSONCXX_DISABLE_WARNING
-            BSONCXX_CONCAT
-            BSONCXX_CONCAT_IMPL
-            bsoncxx_cxx14_constexpr
-            BSONCXX_FORCE_SEMICOLON
-            BSONCXX_FWD
-            BSONCXX_PRAGMA
-            _bsoncxxPragma
-            BSONCXX_STRINGIFY
-            BSONCXX_STRINGIFY_IMPL
-            BSONCXX_IF_MSVC
-            BSONCXX_IF_GCC
-            BSONCXX_IF_CLANG
-            BSONCXX_IF_GNU_LIKE
-            BSONCXX_RETURNS
         INCLUDE_PATTERNS
             "include/*.hpp" # Public headers.
```

Comment:
Reminder that the macro guard tests do not test generated headers (config.hpp, export.hpp, version.hpp), thus their provision of macros when standalone-included is not a concern (as is the case with `macros.hpp`).